### PR TITLE
added permissive CORS header middleware function

### PR DESCRIPTION
**Pull request by MaxBittker**
Hi again... terribly sorry for causing noise and opening a new issue, I just had ran into a problem squashing the commits. 

previous discussion:
https://github.com/reactjs/react-tutorial/pull/118

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -23,6 +23,17 @@ app.set('port', (process.env.PORT || 3000));
 app.use('/', express.static(path.join(__dirname, 'public')));
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({extended: true}));
+app.use(function(req, res, next) {
+    //set permissive CORS headers
+    res.setHeader('Access-Control-Allow-Origin', '*');
+    res.setHeader('Access-Control-Allow-Methods', 'GET,OPTIONS,HEAD,PUT,POST,DELETE,PATCH');
```
Comment (spicyj) [inline server.js]: Do you really need all of these? I would think that just Access-Control-Allow-Origin is all you need.

Comment (MaxBittker) [inline server.js]: I agree, except I believe that the preflight is also needed for POST? so I could remove the method and headers lines.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS

Comment (MaxBittker) [inline server.js]: after testing, I couldn't recreate this scenario with Firefox or chromium, so i remove the extra statements per your suggestion

**Comment by zpao** [comment]
Squashed and pushed as 3fba2d4e1ee9b5ab5253a0c1cdcdcc9bd7879f7b. Thanks!

**Comment by MaxBittker** [comment]
thanks for maintaining the tutorial! enjoyed it.

**Pull request closed without merging**